### PR TITLE
add VoidTransform

### PR DESCRIPTION
**Pull request by jun-he**
To split the PR https://github.com/apache/iceberg/pull/3450, open a new PR for void transform here.

### CODE REVIEW

File: python/src/iceberg/transforms.py [hunk 1]
```diff
@@ -249,6 +249,32 @@ def result_type(self, source: IcebergType) -> IcebergType:
         return StringType()
 
 
+class VoidTransform(Transform):
+    """A transform that always returns None"""
+
+    _instance = None
+
+    def __new__(cls):
```
Comment (Fokko) [inline python/src/iceberg/transforms.py:257]: We could add a test for this:
```python
def test_check_if_identical():
    assert transforms.always_null() is transforms.always_null()
```

File: python/src/iceberg/transforms.py [hunk 2]
```diff
@@ -249,6 +249,32 @@ def result_type(self, source: IcebergType) -> IcebergType:
...
+    def apply(self, value):
+        return None
+
+    def can_transform(self, target: IcebergType) -> bool:
+        return True
+
+    def result_type(self, source: IcebergType) -> IcebergType:
+        return source
+
+    def to_human_string(self, value) -> str:
```
Comment (Fokko) [inline python/src/iceberg/transforms.py:274]:
```suggestion
    def to_human_string(self, Optional[value]) -> str:
```

**Comment by rdblue** [comment]
@jun-he, since #4728 was merged, I think this will fail tests. Can you update please?

File: python/src/iceberg/transforms.py [hunk 3]
```diff
@@ -249,6 +249,32 @@ def result_type(self, source: IcebergType) -> IcebergType:
...
+
+    def __new__(cls):
+        if cls._instance is None:
+            cls._instance = super(VoidTransform, cls).__new__(cls)
+        return cls._instance
+
+    def __init__(self):
+        super().__init__("void", "transforms.always_null()")
+
+    def apply(self, value):
```
[the remark this answers is not in the document]
Comment (jun-he) [reply python/src/iceberg/transforms.py:265]: Thanks for the comment. Updated.

File: python/src/iceberg/transforms.py (hunk 1 above)
Comment (jun-he) [reply python/src/iceberg/transforms.py:257]: 👍  Updated the test.

**Comment by jun-he** [comment]
@rdblue @Fokko I rebased it and updated the PR accordingly. Can you please take a look? Thanks.

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks, @jun-he!

**Pull request merged**